src/client.rs

```diff
@@ -88,40 +85,23 @@ impl Client {
 
     /// Reads events from a given stream. The reading can be done forward and
     /// backward.
-    pub async fn read_stream<Count>(
+    pub async fn read_stream(
         &self,
         stream_name: impl AsRef<str>,
         options: &ReadStreamOptions,
-        count: Count,
-    ) -> crate::Result<Count::Selection>
-    where
-        Count: ToCount,
-    {
-        let stream = commands::read_stream(
-            self.client.clone(),
-            options,
-            stream_name,
-            count.to_count() as u64,
-        )
-        .await?;
-
-        Ok(count.select(stream).await)
+        count: usize,
```

Comment:
The documentation site and most[^1] other clients name this option `maxCount`.  Also, they have it optional, with the default being reading all events in the stream. 

Is there a reason not to move it to `ReadStreamOptions` and have it part of it's default as `usize::MAX`?

[^1]: I only checked dotnet and node, so maybe not most.

Reply:
Fair point 👍

Reply:
Done